packages/plugin-hardhat/src/utils/deploy.ts

```diff
@@ -1,13 +1,14 @@
 import type { Deployment, RemoteDeploymentId } from '@openzeppelin/upgrades-core';
 import debug from './debug';
 import type { ethers, ContractFactory } from 'ethers';
-import { getContractAddress } from 'ethers/lib/utils';
+import { getCreateAddress } from 'ethers';
 import { HardhatRuntimeEnvironment } from 'hardhat/types';
 import { platformDeploy } from '../platform/deploy';
 import { PlatformDeployOptions, UpgradeOptions } from './options';
+import { getSigner } from './ethers';
 
 export interface DeployTransaction {
-  deployTransaction: ethers.providers.TransactionResponse;
+  deployTransaction: null | ethers.TransactionResponse;
```

Comment:
We don't seem to need this here either.

```suggestion
  deployTransaction: ethers.TransactionResponse;
```